internal/repository/prisma/step_run.go

```diff
@@ -527,7 +527,7 @@ func (s *stepRunEngineRepository) assignStepRunToWorkerAttempt(ctx context.Conte
 
 	if err != nil {
 		if errors.Is(err, pgx.ErrNoRows) {
-			return nil, repository.ErrNoWorkerAvailable
+			s.l.Warn().Err(err).Msg("no rows returned from worker assign")
```

Comment:
do we still want to return, we're just going to err on ln 533

Reply:
Right, we'll error but I think that's somewhat expected - we should bubble up this error